plot/Makefile

```diff
@@ -0,0 +1,132 @@
+	@echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@"
+	@grep 'best model' $(MODEL_LOG) \
+		| sed -e 's/^.*\///' \
+		| sed -e 's/\.checkpoint.*$$/\t\t\t/' \
+		| sed -e 's/_/\t/g' >>  "$@"
+
+# Make TSV with Eval CER.
+$(TMP_EVAL): $(MODEL_LOG)
+	@echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@"
+	@grep 'Eval Char' $(MODEL_LOG) \
+		| sed -e 's/^.*[0-9]At iteration //' \
+		| sed -e 's/,.* Eval Char error rate=/\t\t/'  \
+		| sed -e 's/, Word.*$$/\t\t/' \
+		| sed -e 's/^/\t\t/' >>  "$@"
+
+# Validation log is generated only when CER falls below $(VALIDATE_CER)].9, so check for file existence.
+# Extract Filename and CER from Validate log for building Validate TSV file.
+# Make TSV with Validate CER.
+$(TMP_VALIDATE): $(TMP_FAST_LOG)
+	@if [ ! -s $(TMP_FAST_LOG) ]; \
+	then \
+		echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@" ;\
+	else \
+		grep -E "$(VALIDATE_LIST).log$$|iteration" $(TMP_FAST_LOG) > $(TMP_VALIDATE_LOG) ;\
+		echo "Name	CheckpointCER	LearningIteration	TrainingIteration	EvalCER	IterationCER	ValidationCER" > "$@" ;\
+		sed 'N;s/\nAt iteration 0, stage 0, /At iteration 0, stage 0, /;P;D'  $(TMP_VALIDATE_LOG) \
+			| grep 'Eval Char' \
+			| sed -e "s/.$(VALIDATE_LIST).log.*Eval Char error rate=/\t\t\t/" \
+			| sed -e 's/, Word.*$$//' \
+			| sed -e 's/\(^.*\)_\([0-9].*\)_\([0-9].*\)_\([0-9].*\)\t/\1\t\2\t\3\t\4\t/g' >> "$@" ;\
+	fi;
+
+# Build fast traineddata file list with CER in range [0-VALIDATE_CER].[0-9].
+FAST_DATA_FILES := $(shell find ../data/$(MODEL_NAME)/tessdata_fast/ -type f -name $(MODEL_NAME)_[0-$(VALIDATE_CER)].[0-9]*_*.traineddata | sort -n -r)
+
+# Build validate log files list based on above traineddata list.
+FAST_LOG_FILES := $(subst tessdata_fast,tessdata_fast,$(patsubst %.traineddata,%.$(VALIDATE_LIST).log,$(FAST_DATA_FILES)))
+
+#Note: This does not find the new traineddata files from current run. 
+#	Hence `make` needs to be run twice to generate new validate.log files.
+
+$(FAST_LOG_FILES): %.$(VALIDATE_LIST).log: %.traineddata
+	OMP_THREAD_LIMIT=1 time -p lstmeval  \
+		--verbosity=0 \
+		--model $< \
+		--eval_listfile $(TMP_VALIDATE_LIST) 2>&1 | grep "^At iteration" > $@
+
+# Concatenate all validate.log files along with their filenames so as to include iteration number for TSV.
+$(TMP_FAST_LOG): $(FAST_LOG_FILES)
+	@for i in $^; do \
+		echo Filename : "$$i";echo;cat "$$i"; \
+	done > $@
+
+# Combine TSV files with all required CER values, generated from training log and validation logs. Plot.
+$(TSV_VALIDATE_CER): $(TMP_FAST_LOG) $(TMP_100_ITERATIONS) $(TMP_CHECKPOINT) $(TMP_EVAL) $(TMP_VALIDATE) 
+	@cat $(TMP_100_ITERATIONS) $(TMP_CHECKPOINT) $(TMP_EVAL) $(TMP_VALIDATE) > "$@"
+	python plot-eval-validate-cer.py -m $(MODEL_NAME) -v $(VALIDATE_LIST) -y $(Y_MAX_CER)
+	@rm tmp-$(MODEL_NAME)-$(VALIDATE_LIST)*.*
```

Comment:
Should be placed into a separate (phony) rule (like `clean`), and instead of redefining the filenames implicitly (`tmp-*`), re-use the actual above definitions.